src/PartnerRegistrar.ts

```diff
@@ -84,18 +128,33 @@ export class PartnerRegistrar {
    * @param addr the address to set for the name resolution
    * @returns {CommitmentResult} the result of the commitment
    */
-  async commit (label: string, owner: string, duration: BigNumber, addr?: string): Promise<{ secret: string, hash: string }> {
-    const secret = generateSecret()
-    const hash = await this.partnerRegistrar.makeCommitment(hashLabel(label), owner, secret, duration, addr ?? owner)
-    const makeCommitmentTransaction = await this.partnerRegistrar.commit(hash, this.partnerAddress)
-    await makeCommitmentTransaction.wait()
+  commit (label: string, owner: string, duration: BigNumber, addr?: string): Promise<{ secret: string; hash: string }> {
+    return this.commitOp(label, owner, duration, addr).execute()
+  }
 
-    return {
-      secret,
-      hash
+  estimateGas <T extends AcceptedOperationNames> (operationName: T, ...args: AcceptedArgs<T>): Promise<BigNumber> {
+    switch (operationName) {
+      case 'commit':
+        return this.commitOp(args[0], args[1] as string, args[2] as BigNumber, args[3] as string).estimateGas()
+
+      case 'register':
+        return this.registerOp(args[0], args[1] as string, args[2] as string, args[3] as BigNumber, args[4] as BigNumber, args[5] as string).estimateGas()
+
+      case 'renew':
+        return this.renewOp(args[0], args[1] as BigNumber, args[2] as BigNumber).estimateGas()
+
+      case 'commitAndRegister':
+        return this.commitAndRegisterOp(args[0], args[1] as string, args[2] as BigNumber, args[3] as BigNumber, args[4] as string, args[5] as string).estimateGas()
+      default:
+        return this.commitOp(args[0], args[1] as string, args[2] as BigNumber, args[3] as string).estimateGas()
```

Comment:
We should return an error for operations that are not supported

Reply:
yes. I dont know how far typescript takes care of validation tho. I'm guessing it autogenerates the code that validates for the string values once the dist transpilation happens.